pkg/scheduler/core/generic_scheduler.go

```diff
@@ -137,7 +137,7 @@ type ScheduleAlgorithm interface {
 	// the pod by preempting lower priority pods if possible.
 	// It returns the node where preemption happened, a list of preempted pods, a
 	// list of pods whose nominated node name should be removed, and error if any.
-	Preempt(*v1.Pod, error) (selectedNode *v1.Node, preemptedPods []*v1.Pod, cleanupNominatedPods []*v1.Pod, err error)
+	Preempt(*framework.PluginContext, *v1.Pod, error) (selectedNode *v1.Node, preemptedPods []*v1.Pod, cleanupNominatedPods []*v1.Pod, err error)
```

Comment:
pod and then pluginContext? I prefer we are consistent with `Schedule`

Reply:
I prefer the keep the context as the first argument, there is a chance it would conform `context.Context`.

Reply:
@alculquicondor 
FYI:https://golang.org/pkg/context/
_The Context should be the first parameter_

Reply:
Ok. I will do that. :)

Reply:
IMO we could accept this since we have already reached a consensus on conforming the Context in https://github.com/kubernetes/kubernetes/issues/81433 and https://github.com/kubernetes/kubernetes/pull/82072 is making the change.